### PR TITLE
[3.0] Remove more deprecated interfaces in Form and Validator

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php [hunk 1]
```diff
@@ -25,9 +25,7 @@ class DependencyInjectionExtension implements FormExtensionInterface
     private $guesser;
     private $guesserLoaded = false;
 
-    public function __construct(ContainerInterface $container,
-        array $typeServiceIds, array $typeExtensionServiceIds,
-        array $guesserServiceIds)
+    public function __construct(ContainerInterface $container, array $typeServiceIds, array $typeExtensionServiceIds, array $guesserServiceIds)
```
Comment (Tobion) [inline src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php]: if this is done, then not only in master

**Comment by Tobion** [comment]
:+1: apart from the above comment

File: src/Symfony/Component/Form/Form.php [hunk 2]
```diff
@@ -506,10 +505,6 @@ public function handleRequest($request = null)
      */
     public function submit($submittedData, $clearMissing = true)
     {
-        if ($submittedData instanceof Request) {
-            @trigger_error('Passing a Symfony\Component\HttpFoundation\Request object to the '.__CLASS__.'::bind and '.__METHOD__.' methods is deprecated since 2.3 and will be removed in 3.0. Use the '.__CLASS__.'::handleRequest method instead. If you want to test whether the form was submitted separately, you can use the '.__CLASS__.'::isSubmitted method.', E_USER_DEPRECATED);
-        }
```
Comment (fabpot) [inline src/Symfony/Component/Form/Form.php]: We must also remove support for when a Request is passed, I don't see where it's done here.

Comment (Tobion) [inline src/Symfony/Component/Form/Form.php]: There is/was a listener that transforms the data.

Comment (Tobion) [inline src/Symfony/Component/Form/Form.php]: https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Form/Extension/HttpFoundation/EventListener/BindRequestListener.php
its already removed in master

Comment (fabpot) [inline src/Symfony/Component/Form/Form.php]: thanks for the investigation

File: src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php (hunk 1 above)
Comment (fabpot) [inline src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php]: I will let @nicolas-grekas submit a PR for this on 2.3

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

File: src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php]: See https://github.com/symfony/symfony/pull/16084

**Pull request merged**